src/porepy/numerics/fv/mpfa.py

```diff
@@ -463,6 +463,15 @@ def _local_discr(self, g, k, bnd, eta=None, inverter="numba", apertures=None):
         # discretization.
         subcell_topology = fvutils.SubcellTopology(g)
 
+        # Below, the boundary conditions should be defined on the subfaces.
+        if bnd.num_faces == subcell_topology.num_subfno_unique:
+            # The boundary conditions is already given on the subfaces
+            subface_rhs = True
+        else:
+            # If bnd is not already a sub-face_bound we extend it
+            bnd = pp.fvutils.boundary_to_sub_boundary(bnd, subcell_topology)
```

Comment:
When I see how much this cleaned up below, I really like it.

Reply:
exactly! It is basically just committing to the idea of subfaces, also on the boundary